### ISSUE TITLE
Improve usage and handling of script 'docstrings'

### ISSUE THREAD

**Issue by PeridexisErrant** [DFHack/dfhack#823]
Thanks to Sphinx and some custom code in `conf.py`, documentation for scripts is stored in their source files and built into the docs at compile time.

However, there are two changes I intend to make (pending a some of the pull requests that may clash):

1. Change the marker style for Lua to avoid the use of `=begin` and `=end`.  This will facilitate use of the same string, without any duplication, in online docs, as a top-of-file explanation, and as the help text in the DFHack terminal.

2. Consequential changes, eg updating docs and example scripts, merging any separate helpstrings with docstrings as appropriate, etc.

May cause minor conflict with #722 depending on implementation timing.

Requires #723, as I'm not going to do separate pull requests against all the script subrepos and deal with the pain of it not being synced again!

**Comment by PeridexisErrant** [comment]
[00e1186d](https://github.com/PeridexisErrant/dfhack/commit/00e1186d186e697cd715c33b81d2cc3e3e0ee9cd) has the changes to `conf.py`, and contains a script to rewrite the in-script markers too.

As noted above, I now consider this issue blocked by #723 as it's simply too painful and error-prone to go through six repos making changes that should be atomic.

**Comment by PeridexisErrant** [comment]
Note:  must also update [the docs about script docs](https://dfhack.readthedocs.org/en/stable/docs/Contributing.html#documentation-standards).

**Comment by PeridexisErrant** [comment]
`help $script` should print the full docs, not just the first line comment for `ls`.  This probably means we can leave the ruby-style tokens in, and process strings identically.  We should alias any old per-script help options to `man $script`, and integrate the old help.

Related:  it would be nice to have a single-source-of-truth for plugin docs too, but I'm not sure how to do this.  Either Sphinx has to pull out the text from C++ source at docs-build time, or the plugin has to pull them in from somewhere at compile time.

**Comment by PeridexisErrant** [comment]
Should fix typo in `hfs-pit` docs at the same time ([report](http://www.bay12forums.com/smf/index.php?topic=139553.msg6945896#msg6945896))
